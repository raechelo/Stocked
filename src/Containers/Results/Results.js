import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import { connect } from 'react-redux';
import RecipeCard from '../../Components/RecipeCard/RecipeCard';
import Card from '../../Components/Card/Card';
import Instructions from '../../Components/Instructions/Instructions';

export class Results extends Component {
  constructor() {
    super();
    this.state = {
      isExpanded: false,
      exId: 0
    }
  }

  handleClick = (e) => {
    const { id } = e.target
    this.setState({isExpanded: !this.state.isExpanded, exId: id})
  }

  render() {
    const displayCards = this.props.recipes.map(r => (<Card {...r} key={r.id} onClick={() => this.handleClick} />))

    const displayRecipe = this.props.recipes.find(r => r.id === this.state.exId)

    return (
      <section className="Results">
        {this.props.recipes.length ? displayCards : <Instructions />}
        {this.state.isExpanded && <RecipeCard {...displayRecipe} />}
      </section>
    )
  }
}

Results.propTypes = {
  recipes: PropTypes.array
}

export const mapStateToProps = state => ({
  recipes: state.recipes
})

export default connect(mapStateToProps)(Results);