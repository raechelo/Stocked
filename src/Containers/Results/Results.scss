.Results {
  padding: 1rem;
}