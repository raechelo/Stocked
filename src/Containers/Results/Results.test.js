import { Results } from './Results';
import { shallow } from 'enzyme';
import React from 'react';

describe('Results', () => {

  let wrapper;
  let recipes = [{title: 'Mac&Cheese'}]

  beforeEach(() => {
    wrapper = shallow(<Results recipes={recipes} />)
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();

    wrapper.setState({recipes: []})
    expect(wrapper).toMatchSnapshot();
  });
});