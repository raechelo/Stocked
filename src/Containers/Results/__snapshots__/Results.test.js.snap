// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results should match the snapshot with all the data passed in 1`] = `ShallowWrapper {}`;

exports[`Results should match the snapshot with all the data passed in 2`] = `ShallowWrapper {}`;
